lib/httpigeon/request.rb

```diff
@@ -67,15 +69,24 @@ def run(method: :get, path: '/', payload: {})
 
     private
 
-    attr_reader :path, :logger, :event_type, :log_filters
+    attr_reader :logger, :event_type, :log_filters
 
     def parse_response
-      JSON.parse(response_body).with_indifferent_access
-    rescue JSON::ParserError
+      parsed_body = response_body.is_a?(String) ? JSON.parse(response_body) : response_body
+
+      case parsed_body
+      when Hash
+        parsed_body.with_indifferent_access
+      when Array
+        parsed_body.map(&:with_indifferent_access)
```

Comment:
this will still break in the case where the elements of the top-level array are themselves arrays or non-Object in general

for example, it would break on this valid json response in several different ways:
```json
[
  {
    "foo": "bar"
  },
  1,
  "foobar",
  true,
  null,
  [
    {
      "inner": "object"
    },
    1,
    null,
    []
  ]
]
```

consider modifying the contract of the library to only return string-type json 

or, consider a safe recursive "deep with indifferent access." maybe something like this 

```ruby
def deep_with_indifferent_access(obj)
  case obj
  when Hash
    obj.each_with_object({}) do |(key, value), memo|
      memo[key] = deep_with_indifferent_access(value)
    end.with_indifferent_access
  when Array
    obj.map { |item| deep_with_indifferent_access(item) }
  else
    obj
  end
end

test = JSON.parse('[{"foo": "bar"},1,"foobar",true,null,[{"inner": "object"},1,null,[]]]')
result = deep_with_indifferent_access(test)
# => [{"foo"=>"bar"}, 1, "foobar", true, nil, [{"inner"=>"object"}, 1, nil, []]]

result[0][:foo]
# => "bar"

result[5][0][:inner]
# => "object"
```

Reply:
That's what the `NoMethodError` rescue is for hehe. "I tried to help but your thing is weird so I'm leaving it as-is, enjoy"